keras_cv/applications/object_detection/retina_net/__internal__/utils/anchor_box.py

```diff
@@ -0,0 +1,123 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tensorflow as tf
+
+import keras_cv
+
+
+class AnchorBox(tf.keras.layers.Layer):
+    """Generates anchor boxes.
+
+    This class has operations to generate anchor boxes for feature maps at
+    strides `[8, 16, 32, 64, 128]`.
+
+    Args:
+        bounding_box_format:   The format of bounding boxes of input dataset. Refer
+            https://github.com/keras-team/keras-cv/blob/master/keras_cv/bounding_box/converters.py
+            for more details on supported bounding box formats.
+        aspect_ratios: A list of float values representing the aspect ratios of
+            the anchor boxes at each location on the feature map
+        scales: A list of float values representing the scale of the anchor boxes
+            at each location on the feature map.
+        num_anchors: The number of anchor boxes at each location on feature map
+        areas: A list of float values representing the areas of the anchor
+            boxes for each feature map in the feature pyramid.
+        strides: A list of float value representing the strides for each feature
+            map in the feature pyramid.
+
+    To customize these values, please fork the RetinaNet model repo.
+    """
+
+    def __init__(
+        self,
+        bounding_box_format,
+        aspect_ratios=None,
+        scales=None,
+        strides=None,
+        areas=None,
+        **kwargs
+    ):
+        super().__init__(**kwargs)
+        self.bounding_box_format = bounding_box_format
+        self.aspect_ratios = aspect_ratios or [0.5, 1.0, 2.0]
```

Comment:
Should we have those as default value in the method signature?

Reply:
This particular implementation is low-fidelity and will be replaced by Zhenyu's in the near future as a series of 3 PRs introducing reusable components to the end to end OD pipeline.

Lets leave fine-tuning this anchor box class to that PR, and focus on RetinaNet & ObjectDetectionLoss in this PR.